### ISSUE TITLE
Google php cloud function issue, not working

### ISSUE THREAD

**Issue by poyeshsoftware** [GoogleCloudPlatform/functions-framework-php#170]
I can't use php cloud function to listen to an event with php event listener:

my code:

```
<?php
use Psr\Http\Message\ServerRequestInterface;
use Google\CloudFunctions\FunctionsFramework;
use Predis\Client;
use Google\Cloud\PubSub\PubSubClient;
use CloudEvents\V1\CloudEventInterface;


// Register the function with Functions Framework.
// This enables omitting the `FUNCTIONS_SIGNATURE_TYPE=cloudevent` environment
// variable when deploying. The `FUNCTION_TARGET` environment variable should
// match the first parameter.
FunctionsFramework::cloudEvent('UnmodifiedKeyEvent', 'convertMessage');

function convertMessage(CloudEventInterface $event): void
{
      $cloudEventData = $event->getData();
      $messageData = base64_decode($cloudEventData['message']['data']);

    $message = json_decode($messageData, true);
    error_log("message data: " . $messageData);
    
    $redis = new Client([
    'scheme' => 'tcp',
    'host'   => 'x.x.x.x',
    'port'   => 6379,
    ]);
    
    $isBot = $redis->get($message['ip']) !== null;
    $isConversion = strpos($message['url'], 'thank-you') !== false;

    $convertedMessage = [
        'key' => $message['key'],
        'timestamp' => date(DATE_ATOM, intval($message['timestamp'] / 1000)),
        'isBot' => $isBot,
        'isConversion' => $isConversion,
    ];

    $pubsub = new PubSubClient();
    $topic = $pubsub->topic('ModifiedKeyEvent');
    $topic->publish(['data' => json_encode($convertedMessage)]);
}
```

Error that I recieve:

```
{
  "textPayload": "2023/09/30 15:31:00 [error] 19#19: *5 FastCGI sent in stderr: \"PHP message: PHP Fatal error:  Uncaught LogicException: Your function must have \"Google\\CloudFunctions\\CloudEvent\" as the typehint for the first argument in /workspace/vendor/google/cloud-functions-framework/src/CloudEventFunctionWrapper.php:65\nStack trace:\n#0 /workspace/vendor/google/cloud-functions-framework/src/CloudEventFunctionWrapper.php(76): Google\\CloudFunctions\\CloudEventFunctionWrapper->throwInvalidFirstParameterException()\n#1 /workspace/vendor/google/cloud-functions-framework/src/FunctionValidationTrait.php(43): Google\\CloudFunctions\\CloudEventFunctionWrapper->validateFirstParameter(Object(ReflectionParameter))\n#2 /workspace/vendor/google/cloud-functions-framework/src/CloudEventFunctionWrapper.php(59): Google\\CloudFunctions\\CloudEventFunctionWrapper->validateFunctionSignature(Object(ReflectionFunction))\n#3 /workspace/vendor/google/cloud-functions-framework/src/Invoker.php(59): Google\\CloudFunctions\\CloudEventFunctionWrapper->__construct('convertMessage', false)\n#4 /workspace/vendor/google/cloud\" while reading response header from upstream, client: 169.254.1.1, server: , request: \"POST / HTTP/1.1\", upstream: \"fastcgi://unix:/tmp/google-config/app.sock:\", host: \"convertmessage-sa7f4rzqoa-ey.a.run.app\"",
  "resource": {
    "type": "cloud_run_revision",
    "labels": {
      "project_id": "ipfilterflow",
      "service_name": "convertmessage",
      "configuration_name": "convertmessage"
    }
  }
}
```

**Comment by poyeshsoftware** [comment]
my composer.json:

```
{
    "name": "gcp/cloud-functions-convert",
    "type": "project",
    "description": "A Google Cloud Function to convert message",
    "require": {
            "php": ">=7.4",
            "guzzlehttp/psr7": "^1.7|^2.0",
            "psr/http-message": "^1.0",
            "cloudevents/sdk-php": "^1.0",
            "google/cloud-functions-framework": "^1.1.0",
            "google/cloud-pubsub": "^1.32",
            "predis/predis": "^1.1"
    }
}
```

**Comment by poyeshsoftware** [comment]
I use it to listen to pubsub event to modify the message.

**Comment by poyeshsoftware** [comment]
The issue was fixed after I updated it to php 8.2
